lib/common/http1client.c

```diff
@@ -25,53 +25,18 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <sys/un.h>
-#include "picohttpparser.h"
-#include "h2o.h"
-
-struct st_h2o_http1client_private_t {
-    h2o_http1client_t super;
-    union {
-        h2o_http1client_connect_cb on_connect;
-        h2o_http1client_head_cb on_head;
-        h2o_http1client_body_cb on_body;
-    } _cb;
-    h2o_url_t *_origin;
-    h2o_timeout_entry_t _timeout;
-    int _method_is_head;
-    int _do_keepalive;
-    union {
-        struct {
-            size_t bytesleft;
-        } content_length;
-        struct {
-            struct phr_chunked_decoder decoder;
-            size_t bytes_decoded_in_buf;
-        } chunked;
-    } _body_decoder;
-    h2o_socket_cb reader;
-    h2o_http1client_proceed_req_cb proceed_req;
-    char _chunk_len_str[(sizeof(H2O_UINT64_LONGEST_HEX_STR) - 1) + 2 + 1]; /* SIZE_MAX in hex + CRLF + '\0' */
-    h2o_buffer_t *_body_buf;
-    h2o_buffer_t *_body_buf_in_flight;
-    unsigned _is_chunked : 1;
-    unsigned _body_buf_is_done : 1;
-};
-
-static void close_client(struct st_h2o_http1client_private_t *client)
-{
-    if (client->super.sock != NULL) {
-        if (client->super.sockpool.pool != NULL && client->_do_keepalive) {
+#include "h2o/httpclient_internal_h1.h"
+#include "h2o/token.h"
 
+static void close_client(struct st_h2o_http1client_t *client)
+{
+    if (client->sock != NULL) {
```

Comment:
Do we still need this `if` now that we construct http1client after the connection is established?

Reply:
yes, I think it's needed because there's a possibility that `steal_socket` is called, which set client->socket NULL, before `close_client` is called.